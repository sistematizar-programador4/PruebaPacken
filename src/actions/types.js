export const SEARCH_PRODUCT = "SEARCH_PRODUCT";
export const FILTER_PRODUCT_BY = "FILTER_PRODUCT_BY";
export const ADD_TO_CART = 'ADD_TO_CART';
export const EDIT_QUANTITY = 'EDIT_QUANTITY';
export const REMOVE_FROM_CART = 'REMOVE_FROM_CART';
export const EMPTY_CART = 'EMPTY_CART';
export const ADD_ORDER = 'ADD_ORDER';
