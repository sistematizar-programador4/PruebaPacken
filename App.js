
import { createStackNavigator } from "react-navigation-stack";
import { createAppContainer } from "react-navigation";
import CategoryList from "./src/components/categories/list.component";
import CategorySubList from "./src/components/categories/sublist.component";
import ProductList from "./src/components/products/list.component";
import Checkout from "./src/components/cart/checkout.component"

const AppStack = createStackNavigator({
  categoryList: CategoryList,
  categorySubList: CategorySubList,
  productList: ProductList,
  checkout: Checkout
});

export default createAppContainer(AppStack);