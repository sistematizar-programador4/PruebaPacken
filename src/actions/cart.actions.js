import { ADD_TO_CART, REMOVE_FROM_CART, EMPTY_CART } from './types';

export const addToCart = (item) => ({
        type: ADD_TO_CART,
        payload: item
})

export const removeItem = (item) => ({
        type: REMOVE_FROM_CART,
        payload: item
})

export const emptyCart = () => ({
        type: EMPTY_CART
})