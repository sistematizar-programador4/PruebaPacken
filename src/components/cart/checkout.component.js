import React, { Component } from 'react';
import { connect } from 'react-redux';
import CheckoutItems from '../cart/checkoutItems.component';
import Cart from "../cart/cart.component";

export class Checkout extends Component {

 static navigationOptions = ({navigation}) => {
    return {
      headerTitle: 'Checkout',
      headerRight: <Cart navigation={navigation}/>,
      headerTintColor: "white",
      headerStyle: {
        backgroundColor: "#845cc3"
      }
    }
  }
    render() {
    	const { cartItems, navigation, cartTotal } = this.props;
        return (
            <CheckoutItems cartItems={cartItems} cartTotal={cartTotal} navigation={navigation}/>
        );
    }
}

const mapStateToProps = state => {
    return {
        cartItems: state.cartReducer.cart,
        cartTotal: state.cartReducer.total
    };
};

export default connect(
    mapStateToProps
)(Checkout);