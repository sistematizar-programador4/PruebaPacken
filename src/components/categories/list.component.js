import React, { Component } from "react";
import { StyleSheet, FlatList, View } from "react-native";
import { ListItem, Icon } from "react-native-elements";
import { connect } from "react-redux";
import Cart from "../cart/cart.component";

class CategoriesList extends Component {
  static navigationOptions = ({navigation}) => {
    return {
      headerTitle: 'Products',
      headerRight: <Cart navigation={navigation}/>,
      headerTintColor: "white",
      headerStyle: {
        backgroundColor: "#845cc3"
      }
    }
  };
  FlatListItemSeparator = () => {
    return (
      <View
        style={{
          height: 5,
        }}
      />
    );
  }
  render() {
    return (
      <FlatList
        style={styles.listContainer}
        data={this.props.categoryList}
        ItemSeparatorComponent = { this.FlatListItemSeparator }
        keyExtractor={(item, index) => item.id.toString()}
        renderItem={data => (
            <ListItem
              title={data.item.name}
              rightIcon={
                <Icon
                  name="arrow-forward"
                  size={18}
                  onPress={() => (this.props.navigation.navigate("categorySubList",{categories:data.item.sublevels}) || null)}
                />
              }
            />
          )}
      />
    );
  }
}

const styles = StyleSheet.create({
  listContainer: {
    backgroundColor: "#dce2ff",
    padding: 16
  },
  listText: {
    fontSize: 30
  }
});

const mapStateToProps = state => {
  return {
    categoryList: state.categoriesReducer.categoryList.categories
  };
};

export default connect(mapStateToProps)(CategoriesList);