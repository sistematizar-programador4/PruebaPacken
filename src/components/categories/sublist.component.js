import React, { Component } from "react";
import { StyleSheet, FlatList, View } from "react-native";
import { ListItem, Icon } from "react-native-elements";
import Cart from "../cart/cart.component";

class CategoriesSubList extends Component {
  static navigationOptions = ({navigation}) => {
    return {
      headerTitle: 'Products',
      headerRight: <Cart navigation={navigation}/>,
      headerTintColor: "white",
      headerStyle: {
        backgroundColor: "#845cc3"
      }
    }
  }
  FlatListItemSeparator = () => {
    return (
      <View
        style={{
          height: 5,
        }}
      />
    );
  };

  FlatListIcon = (item) => {
    if(item.sublevels){
        return (
            <Icon
                name="arrow-forward"
                size={18}
                onPress={() => this.props.navigation.navigate("categorySubList",{categories:item.sublevels})}
            />
        );
    }else{
        return (
            <Icon
                name="restaurant"
                size={18}
                onPress={() => this.props.navigation.navigate("productList",{sublevel_id: item.id,category_name: item.name})}
            />
        );
    }
  };

  render() {
      return (
        <FlatList
            style={styles.listContainer}
            data={this.props.navigation.getParam('categories',null)}
            ItemSeparatorComponent = { this.FlatListItemSeparator }
            keyExtractor={(item, index) => item.id.toString()}
            renderItem={data => (
                <ListItem
                title={data.item.name}
                rightIcon={this.FlatListIcon(data.item)}
                />
            )}
        />
    );
  }
}

const styles = StyleSheet.create({
  listContainer: {
    backgroundColor: "#dce2ff",
    padding: 16
  },
  listText: {
    fontSize: 30
  }
});

export default CategoriesSubList;