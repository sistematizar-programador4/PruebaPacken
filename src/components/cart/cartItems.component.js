import React, { Component } from 'react';
import { 
    View, 
    Text,
    StyleSheet,
    TouchableOpacity,
    Alert 
} from 'react-native';
import { Icon } from "react-native-elements";

import { connect } from 'react-redux';
import { removeItem,addToCart } from '../../actions/cart.actions';

class CartItems extends Component {

    state = {
        activeRowKey: null
    }

    render() {
        const { item, index } = this.props;
        return (
            <View style={styles.container}>
                <View style={styles.productDes}>
                    <Text style={styles.text}>{item.name}</Text>
                    <Text style={styles.text}>{item.price}</Text>
                    <Icon
                        raised
                        name='minus'
                        type='font-awesome'
                        color='#f50'
                        onPress={() => this.props.removeItem(item)} />
                    <Text style={styles.text}>{item.quantity}</Text>
                    <Icon
                        raised
                        name='plus'
                        type='font-awesome'
                        color='#f50'
                        onPress={() => this.props.addToCart({item:item})} />
                </View>
            </View>
    );
    }

}

const styles = StyleSheet.create({
    container:{
        flex: 1,
        backgroundColor: "white",
    },
    productDes: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        margin: 10,
    },
    text: {
        fontSize: 16,
        fontWeight:"bold",
        padding: 10
    }
});

export default connect(null,{removeItem,addToCart})(CartItems);