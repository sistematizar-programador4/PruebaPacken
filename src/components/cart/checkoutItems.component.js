import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
	View,
	Text,
	FlatList,
  StyleSheet,
  ScrollView,
  Button
} from 'react-native';
import { emptyCart } from '../../actions/cart.actions';
import CartItems  from './cartItems.component';
// import CustomerForm from './CustomerForm.component';

class CheckoutItems extends Component {  
  render() {
    const { cartItems } = this.props;
    return (
      <View style={styles.container}>
            <View style={styles.annouc}>
                <Text style={styles.anncText}>Please confirm your order and checkout your cart.</Text>
            </View>
            <View style={styles.ckitems}>
                <FlatList 
                    data={cartItems}
            		renderItem={({item, index}) => <CartItems item={item} index={index} />}
            		keyExtractor={(item) => item.id}
            		ItemSeparatorComponent= {()=> <View style={{height:0.3, backgroundColor:'#34495e90'}}/> }
            	/>
    		</View>
       		<View style={styles.custForm}>
                <ScrollView>
                <Button style={{ alignSelf: 'auto'}} onPress={()=>this.props.emptyCart()} title="Buy"/>  
                </ScrollView> 
        	</View>
       </View>
    );
  }
}

const styles = StyleSheet.create({
  container:{
    flex: 1,
    backgroundColor: "#dce2ff",
  },
  custForm: {
    flex: 1
  },
  ckitems: {
    height: 250
  },
    annouc:{
      padding: 12,
      borderRadius: 5,
      backgroundColor: '#34495e90',
      margin: 10,
      justifyContent: 'center',
      alignItems: 'center'
    },
    text: {
      textAlign: 'center',
      color: 'red',
      fontWeight: "bold",
      fontSize: 22
    },
    anncText:{
        textAlign: 'center',
        color: 'black',
        fontSize: 16,
        fontWeight: "bold"
    }
});

export default connect(null,{emptyCart})(CheckoutItems);