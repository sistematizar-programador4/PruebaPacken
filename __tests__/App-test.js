/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App'
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk'
import configureStore from 'redux-mock-store'
import * as actions from '../src/actions/cart.actions';
import categories from '../src/reducers/categories.reducer'
import products from '../src/reducers/products.reducer'
const middlewares = [thunk]
const mockStore = configureStore(middlewares)
const store = mockStore();

beforeEach(() => {
  store.clearActions();
});

it('INITIAL STATE CATEGORIES', () => {
  const action = { type: 'DUMMY' };
  expect(categories(undefined, action)).toMatchSnapshot();
});

it('INITIAL STATE PRODUCTS', () => {
  const action = { type: 'DUMMY' };
  expect(products(undefined, action)).toMatchSnapshot();
});

it('addToCart ACTION', () => {
  store.dispatch(actions.addToCart({
    "quantity": 308,
    "price": "$8,958",
    "available": false,
    "sublevel_id": 3,
    "name": "aute",
    "id": "58b5a5b1b6b6c7aacc25b3fb"
  }));
  expect(store.getActions()).toMatchSnapshot();
});

it('removeItem ACTION', () => {
  store.dispatch(actions.removeItem({
    "quantity": 308,
    "price": "$8,958",
    "available": false,
    "sublevel_id": 3,
    "name": "aute",
    "id": "58b5a5b1b6b6c7aacc25b3fb"
  }));
  expect(store.getActions()).toMatchSnapshot();
});

it('emptyCart ACTION', () => {
  store.dispatch(actions.emptyCart());
  expect(store.getActions()).toMatchSnapshot();
});