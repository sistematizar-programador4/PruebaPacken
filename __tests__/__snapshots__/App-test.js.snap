// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INITIAL STATE CATEGORIES 1`] = `
Object {
  "categoryList": Object {
    "categories": Array [
      Object {
        "id": 1,
        "name": "Bebidas",
        "sublevels": Array [
          Object {
            "id": 1,
            "name": "Gaseosas",
            "sublevels": Array [
              Object {
                "id": 2,
                "name": "Con azúcar",
              },
              Object {
                "id": 3,
                "name": "Sin azúcar",
              },
            ],
          },
        ],
      },
      Object {
        "id": 2,
        "name": "Desayunos",
        "sublevels": Array [
          Object {
            "id": 4,
            "name": "Fake 1",
            "sublevels": Array [
              Object {
                "id": 5,
                "name": "Fake 2",
              },
              Object {
                "id": 6,
                "name": "Fake 3",
                "sublevels": Array [
                  Object {
                    "id": 7,
                    "name": "Fake 4",
                  },
                ],
              },
            ],
          },
        ],
      },
      Object {
        "id": 8,
        "name": "Almuerzos",
        "sublevels": Array [
          Object {
            "id": 9,
            "name": "Fake 5",
          },
          Object {
            "id": 10,
            "name": "Fake 6",
          },
        ],
      },
      Object {
        "id": 11,
        "name": "Vinos",
        "sublevels": Array [
          Object {
            "id": 12,
            "name": "Fake 8",
          },
          Object {
            "id": 13,
            "name": "Fake 9",
          },
        ],
      },
    ],
  },
}
`;

exports[`INITIAL STATE PRODUCTS 1`] = `
Object {
  "productsList": Object {
    "products": Array [
      Object {
        "available": false,
        "id": "58b5a5b1b6b6c7aacc25b3fb",
        "name": "aute",
        "price": "$8,958",
        "quantity": 308,
        "sublevel_id": 3,
      },
      Object {
        "available": true,
        "id": "58b5a5b117bf36cf8aed54ab",
        "name": "mollit",
        "price": "$5,450",
        "quantity": 891,
        "sublevel_id": 3,
      },
      Object {
        "available": false,
        "id": "58b5a5b18607b1071fb5ab5b",
        "name": "eiusmod",
        "price": "$17,001",
        "quantity": 698,
        "sublevel_id": 10,
      },
      Object {
        "available": true,
        "id": "58b5a5b13881e97291384813",
        "name": "proident",
        "price": "$6,864",
        "quantity": 449,
        "sublevel_id": 7,
      },
      Object {
        "available": false,
        "id": "58b5a5b1b82dc20c7dd52260",
        "name": "laboris",
        "price": "$13,253",
        "quantity": 736,
        "sublevel_id": 4,
      },
      Object {
        "available": false,
        "id": "58b5a5b1996384dbbc556718",
        "name": "anim",
        "price": "$10,930",
        "quantity": 850,
        "sublevel_id": 11,
      },
      Object {
        "available": true,
        "id": "58b5a5b17326875fe21aebc1",
        "name": "duis",
        "price": "$16,972",
        "quantity": 644,
        "sublevel_id": 12,
      },
      Object {
        "available": true,
        "id": "58b5a5b1ac8306b5d1d3d5fd",
        "name": "sunt",
        "price": "$6,904",
        "quantity": 722,
        "sublevel_id": 9,
      },
      Object {
        "available": true,
        "id": "58b5a5b1f9ebd51b88636467",
        "name": "reprehenderit",
        "price": "$13,813",
        "quantity": 620,
        "sublevel_id": 8,
      },
      Object {
        "available": true,
        "id": "58b5a5b1a20b39205ba99c50",
        "name": "non",
        "price": "$14,552",
        "quantity": 297,
        "sublevel_id": 1,
      },
      Object {
        "available": false,
        "id": "58b5a5b11f741600608205ca",
        "name": "dolor",
        "price": "$11,042",
        "quantity": 514,
        "sublevel_id": 1,
      },
      Object {
        "available": true,
        "id": "58b5a5b1ef62cb996bb87c45",
        "name": "dolor",
        "price": "$17,532",
        "quantity": 887,
        "sublevel_id": 13,
      },
      Object {
        "available": true,
        "id": "58b5a5b1a089200683925a83",
        "name": "enim",
        "price": "$6,174",
        "quantity": 202,
        "sublevel_id": 13,
      },
      Object {
        "available": false,
        "id": "58b5a5b12d7859f999ecccf1",
        "name": "ad",
        "price": "$1,904",
        "quantity": 700,
        "sublevel_id": 5,
      },
      Object {
        "available": false,
        "id": "58b5a5b1563189cbccc9adc7",
        "name": "ullamco",
        "price": "$14,388",
        "quantity": 343,
        "sublevel_id": 3,
      },
      Object {
        "available": false,
        "id": "58b5a5b18f0d9e833ed59bcd",
        "name": "mollit",
        "price": "$6,809",
        "quantity": 435,
        "sublevel_id": 6,
      },
      Object {
        "available": true,
        "id": "58b5a5b18041ec3e1f95de07",
        "name": "aute",
        "price": "$9,341",
        "quantity": 104,
        "sublevel_id": 1,
      },
      Object {
        "available": true,
        "id": "58b5a5b1b88a817e7b059cba",
        "name": "magna",
        "price": "$3,793",
        "quantity": 888,
        "sublevel_id": 5,
      },
      Object {
        "available": false,
        "id": "58b5a5b1c9a8564521717e20",
        "name": "aliqua",
        "price": "$13,110",
        "quantity": 888,
        "sublevel_id": 7,
      },
      Object {
        "available": true,
        "id": "58b5a5b1da0b4a52b88fc9f9",
        "name": "irure",
        "price": "$5,827",
        "quantity": 724,
        "sublevel_id": 7,
      },
      Object {
        "available": false,
        "id": "58b5a5b1762f69a24c1b3026",
        "name": "nulla",
        "price": "$14,457",
        "quantity": 119,
        "sublevel_id": 3,
      },
      Object {
        "available": false,
        "id": "58b5a5b10448a50924813126",
        "name": "elit",
        "price": "$1,449",
        "quantity": 104,
        "sublevel_id": 10,
      },
      Object {
        "available": false,
        "id": "58b5a5b1404bba7aeef39ef8",
        "name": "ea",
        "price": "$9,619",
        "quantity": 296,
        "sublevel_id": 5,
      },
      Object {
        "available": true,
        "id": "58b5a5b1abd8ab6045ca08cc",
        "name": "occaecat",
        "price": "$18,849",
        "quantity": 585,
        "sublevel_id": 5,
      },
      Object {
        "available": true,
        "id": "58b5a5b116513baf9ffc5444",
        "name": "quis",
        "price": "$2,843",
        "quantity": 936,
        "sublevel_id": 8,
      },
      Object {
        "available": false,
        "id": "58b5a5b17ab1bf66a2c7b61b",
        "name": "id",
        "price": "$12,902",
        "quantity": 399,
        "sublevel_id": 7,
      },
      Object {
        "available": false,
        "id": "58b5a5b1d99c26ea37cb738c",
        "name": "nostrud",
        "price": "$9,048",
        "quantity": 735,
        "sublevel_id": 3,
      },
      Object {
        "available": false,
        "id": "58b5a5b186acbed348266f1c",
        "name": "reprehenderit",
        "price": "$1,482",
        "quantity": 116,
        "sublevel_id": 1,
      },
      Object {
        "available": true,
        "id": "58b5a5b1b00bd95c14066c8b",
        "name": "voluptate",
        "price": "$5,673",
        "quantity": 649,
        "sublevel_id": 9,
      },
      Object {
        "available": false,
        "id": "58b5a5b18f96294f880390c3",
        "name": "est",
        "price": "$3,767",
        "quantity": 839,
        "sublevel_id": 12,
      },
      Object {
        "available": true,
        "id": "58b5a5b17e64c90c960eb3c7",
        "name": "aliqua",
        "price": "$17,417",
        "quantity": 706,
        "sublevel_id": 4,
      },
      Object {
        "available": false,
        "id": "58b5a5b1bba25641557222f1",
        "name": "eu",
        "price": "$5,023",
        "quantity": 566,
        "sublevel_id": 4,
      },
      Object {
        "available": false,
        "id": "58b5a5b174071d8ab7ca7ee1",
        "name": "Lorem",
        "price": "$9,972",
        "quantity": 331,
        "sublevel_id": 7,
      },
      Object {
        "available": false,
        "id": "58b5a5b1c2935d0602b9ccc5",
        "name": "exercitation",
        "price": "$10,987",
        "quantity": 975,
        "sublevel_id": 10,
      },
      Object {
        "available": true,
        "id": "58b5a5b17b4c47445bf6a234",
        "name": "quis",
        "price": "$7,533",
        "quantity": 340,
        "sublevel_id": 1,
      },
      Object {
        "available": false,
        "id": "58b5a5b1c2b40cb8d3976126",
        "name": "ad",
        "price": "$18,332",
        "quantity": 543,
        "sublevel_id": 11,
      },
      Object {
        "available": true,
        "id": "58b5a5b1b7b2382511968d1e",
        "name": "proident",
        "price": "$15,006",
        "quantity": 687,
        "sublevel_id": 8,
      },
      Object {
        "available": true,
        "id": "58b5a5b11d6d4b1ec9db971c",
        "name": "id",
        "price": "$8,651",
        "quantity": 217,
        "sublevel_id": 1,
      },
      Object {
        "available": false,
        "id": "58b5a5b1e26c8f68850e2ceb",
        "name": "laborum",
        "price": "$8,376",
        "quantity": 466,
        "sublevel_id": 13,
      },
      Object {
        "available": true,
        "id": "58b5a5b13163243d012fb89a",
        "name": "non",
        "price": "$8,732",
        "quantity": 922,
        "sublevel_id": 8,
      },
      Object {
        "available": false,
        "id": "58b5a5b1f3bf845c50e56244",
        "name": "labore",
        "price": "$6,444",
        "quantity": 553,
        "sublevel_id": 10,
      },
      Object {
        "available": true,
        "id": "58b5a5b1d24839bdfbc43800",
        "name": "qui",
        "price": "$9,423",
        "quantity": 859,
        "sublevel_id": 3,
      },
      Object {
        "available": true,
        "id": "58b5a5b14a40d79ef80972fe",
        "name": "eiusmod",
        "price": "$5,653",
        "quantity": 842,
        "sublevel_id": 5,
      },
      Object {
        "available": true,
        "id": "58b5a5b17c36901b87f4f85a",
        "name": "mollit",
        "price": "$15,468",
        "quantity": 967,
        "sublevel_id": 11,
      },
      Object {
        "available": true,
        "id": "58b5a5b12cd2fb555a91652b",
        "name": "consequat",
        "price": "$17,712",
        "quantity": 102,
        "sublevel_id": 3,
      },
      Object {
        "available": true,
        "id": "58b5a5b13bc95b01816c05b1",
        "name": "enim",
        "price": "$2,192",
        "quantity": 240,
        "sublevel_id": 6,
      },
      Object {
        "available": true,
        "id": "58b5a5b12bf26721f9eb7666",
        "name": "occaecat",
        "price": "$18,523",
        "quantity": 994,
        "sublevel_id": 9,
      },
      Object {
        "available": false,
        "id": "58b5a5b1ffde2a73491feb07",
        "name": "elit",
        "price": "$12,909",
        "quantity": 429,
        "sublevel_id": 7,
      },
      Object {
        "available": true,
        "id": "58b5a5b1c60cafd804b3bfc9",
        "name": "sint",
        "price": "$14,970",
        "quantity": 597,
        "sublevel_id": 3,
      },
      Object {
        "available": true,
        "id": "58b5a5b148f556bae5c7d094",
        "name": "in",
        "price": "$9,091",
        "quantity": 898,
        "sublevel_id": 4,
      },
      Object {
        "available": false,
        "id": "58b5a5b1ba6292fc572e9aaf",
        "name": "excepteur",
        "price": "$5,525",
        "quantity": 410,
        "sublevel_id": 11,
      },
      Object {
        "available": false,
        "id": "58b5a5b1f58fc52adfda923b",
        "name": "irure",
        "price": "$15,789",
        "quantity": 176,
        "sublevel_id": 13,
      },
      Object {
        "available": false,
        "id": "58b5a5b137ecc41faaea9240",
        "name": "dolore",
        "price": "$15,502",
        "quantity": 198,
        "sublevel_id": 8,
      },
      Object {
        "available": false,
        "id": "58b5a5b1b64b6d32f2c76262",
        "name": "ad",
        "price": "$10,050",
        "quantity": 835,
        "sublevel_id": 2,
      },
      Object {
        "available": false,
        "id": "58b5a5b12c5ceb44d1462659",
        "name": "pariatur",
        "price": "$6,603",
        "quantity": 144,
        "sublevel_id": 6,
      },
      Object {
        "available": false,
        "id": "58b5a5b13d958af8c892213b",
        "name": "duis",
        "price": "$18,389",
        "quantity": 977,
        "sublevel_id": 13,
      },
      Object {
        "available": false,
        "id": "58b5a5b11a61d1be3c7314ec",
        "name": "ut",
        "price": "$2,056",
        "quantity": 806,
        "sublevel_id": 9,
      },
      Object {
        "available": true,
        "id": "58b5a5b19b7d5b4d3f844ade",
        "name": "aliquip",
        "price": "$4,262",
        "quantity": 639,
        "sublevel_id": 11,
      },
      Object {
        "available": true,
        "id": "58b5a5b147af435dcecd4c7e",
        "name": "aliqua",
        "price": "$10,651",
        "quantity": 767,
        "sublevel_id": 10,
      },
      Object {
        "available": false,
        "id": "58b5a5b1101a59975b4c30b2",
        "name": "id",
        "price": "$16,235",
        "quantity": 758,
        "sublevel_id": 3,
      },
      Object {
        "available": true,
        "id": "58b5a5b13f6e3eef1f16b817",
        "name": "non",
        "price": "$5,982",
        "quantity": 546,
        "sublevel_id": 10,
      },
      Object {
        "available": false,
        "id": "58b5a5b18dbb3f02cb1629f5",
        "name": "reprehenderit",
        "price": "$7,678",
        "quantity": 592,
        "sublevel_id": 9,
      },
      Object {
        "available": false,
        "id": "58b5a5b19212e84560bfd56a",
        "name": "tempor",
        "price": "$17,532",
        "quantity": 945,
        "sublevel_id": 5,
      },
      Object {
        "available": false,
        "id": "58b5a5b1c89c18d8ccc31f5a",
        "name": "non",
        "price": "$9,036",
        "quantity": 307,
        "sublevel_id": 5,
      },
      Object {
        "available": false,
        "id": "58b5a5b136fe73b240c6bd7a",
        "name": "pariatur",
        "price": "$13,615",
        "quantity": 378,
        "sublevel_id": 5,
      },
      Object {
        "available": false,
        "id": "58b5a5b1cc7575c1f6358150",
        "name": "reprehenderit",
        "price": "$7,621",
        "quantity": 346,
        "sublevel_id": 1,
      },
      Object {
        "available": true,
        "id": "58b5a5b1be154bf75f25dec5",
        "name": "aliquip",
        "price": "$14,842",
        "quantity": 150,
        "sublevel_id": 12,
      },
      Object {
        "available": false,
        "id": "58b5a5b144f5f4fa8c040060",
        "name": "veniam",
        "price": "$17,876",
        "quantity": 905,
        "sublevel_id": 7,
      },
      Object {
        "available": false,
        "id": "58b5a5b1f5c3bb7704c16a2d",
        "name": "magna",
        "price": "$6,521",
        "quantity": 717,
        "sublevel_id": 8,
      },
      Object {
        "available": true,
        "id": "58b5a5b1b37cd67f8371b4e3",
        "name": "tempor",
        "price": "$2,504",
        "quantity": 235,
        "sublevel_id": 12,
      },
    ],
  },
}
`;

exports[`addToCart 1`] = `
Array [
  Object {
    "payload": Object {
      "available": false,
      "id": "58b5a5b1b6b6c7aacc25b3fb",
      "name": "aute",
      "price": "$8,958",
      "quantity": 308,
      "sublevel_id": 3,
    },
    "type": "ADD_TO_CART",
  },
]
`;

exports[`removeItem 1`] = `
Array [
  Object {
    "payload": Object {
      "available": false,
      "id": "58b5a5b1b6b6c7aacc25b3fb",
      "name": "aute",
      "price": "$8,958",
      "quantity": 308,
      "sublevel_id": 3,
    },
    "type": "REMOVE_FROM_CART",
  },
]
`;
